Content-check enforcement in enforce-git-delegation.sh: block governed-file push by github-ops

## Background

PR #332 (branch `feature/331-subagent-aware-git-delegation-hook`) shipped
an `enforce-git-delegation.sh` PreToolUse hook with coarse subagent-bypass:
the `f5xc-github-ops:github-ops` subagent was allowed to run any
`git commit`, `git push`, or `gh pr create` in a downstream repo, regardless
of what the changeset touched.

Operator feedback has tightened that policy: **github-ops may run those
delegated operations, but it must not be able to push governed files
(files listed in `.claude/governance.json` → `protected_files`) out of a
downstream repo.** Governed files are owned by `docs-control` and sync
downstream via managed-files sync — any edit to them in a downstream repo
should be rejected at commit/push/PR time so that the only way to change
them is via `docs-control` itself.

## Gap in the current hook

The hook shipped in #332 is binary: either the operation is delegated
(and github-ops is allowed unconditionally), or it's blocked. There is no
inspection of *what* github-ops is about to commit/push/PR. That leaves a
real path for governed files to accidentally leave a downstream repo —
exactly what docs-control's governance model is supposed to prevent.

## What this issue tracks

Add a content-inspection section to `enforce-git-delegation.sh`, appended
after the existing delegation check, that:

- For `git commit`: inspects `git diff --cached --name-only`, plus
  `git diff --name-only` if `-a` / `--all` is set (to cover
  modified-but-unstaged tracked files picked up by `-a`).
- For `git push` / `gh pr create`: inspects
  `git diff --name-only <upstream>..HEAD`, falling back to
  `origin/HEAD` or `origin/main` if the branch has no upstream yet.
- Intersects the computed changeset with `.claude/governance.json`'s
  `protected_files` array.
- On any intersection: exits 2 with a governance-aware BLOCKED message
  listing the violating files and pointing the operator at the
  source-of-truth repo.

The docs-control self-exclusion and the main-session / non-github-ops
blocks from #332 are preserved exactly — this is additive only.

## Acceptance criteria

- `shellcheck` clean on the updated script.
- docs-control is fully self-excluded (every git/GitHub op still allowed
  here — this is the source of truth).
- Main session and non-github-ops subagents are still blocked from
  running `git commit`, `git push`, or `gh pr create` in downstream repos.
- github-ops with **only non-governed** files staged/outgoing → allowed.
- github-ops with **any governed** file staged/outgoing → blocked with a
  clear, actionable BLOCKED message naming the files and
  `source_repo`.
- `-a`, `-am`, `-a -m`, `--all` variants of `git commit` correctly pull
  in modified-but-unstaged tracked files.
- `--amend` and `-m "add feature"` must NOT be mistaken for `-a`.
- `git push` and `gh pr create` use the outgoing diff (upstream..HEAD),
  not just the last commit.
- Hook handles empty stdin, missing `command` field, and missing
  `governance.json` gracefully.

## CI Failure Report

**PR:** #337
**Run:** 24624098408
**Workflow:** Require Linked Issue
**Failed check:** `Check linked issues`

### Failure

```
##[error]No linked issues found. Please add the corresponding issues in
the pull request description.
```

The PR body contains `Closes #336`, but the
`nearform-actions/github-action-check-linked-issues@v1` action resolves
closing-issue links via GitHub's `closingIssuesReferences` relationship,
which GitHub only populates when the PR targets the repository's default
branch (`main`). PR #337 targets
`feature/331-subagent-aware-git-delegation-hook` (a stacked PR on top of
#332), so the action sees zero linked issues even though the body is
correct.

### Why this is not a content defect

- `shellcheck` is clean.
- Pre-commit is green.
- The local 26/26 test suite for the hook passes.
- The only failing check is the `check-linked-issues` action,
  which has a known limitation on non-default-branch PRs.

### Options

1. **Recommended:** Merge #332 to `main` first. Then rebase this branch
   onto `main` and retarget PR #337 from
   `feature/331-subagent-aware-git-delegation-hook` to `main`. The
   linked-issue check will then resolve `Closes #336` correctly.
2. Add the `no-issue` label to #337 to skip this required check
   (the action's `skip-linked-issues-check-label` is set to `no-issue`).
   Use only if the reviewer explicitly accepts that bypass for a
   stacked PR.

Not a code defect. No content change required in the hook.

--- a/.claude/hooks/enforce-git-delegation.sh
+++ b/.claude/hooks/enforce-git-delegation.sh
@@ -1,8 +1,11 @@
 #!/usr/bin/env bash
-# PreToolUse hook: blocks git commit/push and gh pr create from the main
-# session, so all git/GitHub operations are delegated to the
-# f5xc-github-ops:github-ops subagent (per CLAUDE.md).
-# Allows the delegated subagent through via the agent_type stdin field.
+# PreToolUse hook: enforces the docs-control delegation policy.
+# In docs-control itself: everything is allowed (source-of-truth).
+# In downstream repos:
+#   - main session and non-delegated subagents are blocked from running
+#     git commit / git push / gh pr create (must delegate to github-ops).
+#   - the f5xc-github-ops:github-ops subagent may run those operations,
+#     UNLESS the changeset touches a file listed in governance.json.
 # Distributed by docs-control via managed_files sync.
 # Exit 0 = allow, Exit 2 = block (stderr shown to Claude).
 set -euo pipefail
@@ -15,16 +18,8 @@ fi
 INPUT=$(cat)
 
 # ── Self-exclusion: allow all git/GitHub operations in docs-control ─
-# docs-control IS the source of truth; delegation policy applies only
-# to downstream repos that consume it via managed_files sync.
 REMOTE_URL=$(git remote get-url origin 2>/dev/null || echo "")
 if echo "$REMOTE_URL" | grep -q "docs-control"; then
-  exit 0
-fi
-
-# ── Allow the trusted delegated subagent ────────────────────────────
-AGENT_TYPE=$(echo "$INPUT" | jq -r '.agent_type // empty' 2>/dev/null || echo "")
-if [ "$AGENT_TYPE" = "f5xc-github-ops:github-ops" ]; then
   exit 0
 fi
 
@@ -34,16 +29,82 @@ if [ -z "$COMMAND" ]; then
   exit 0
 fi
 
-# ── Block delegated git/gh operations ───────────────────────────────
-BLOCK_REGEX='(^|[^A-Za-z0-9_])(git[[:space:]]+(commit|push)|gh[[:space:]]+pr[[:space:]]+create)([^A-Za-z0-9_]|$)'
+# ── Detect operation type ───────────────────────────────────────────
+IS_COMMIT=0
+IS_PUSH=0
+IS_PR=0
+if [[ "$COMMAND" =~ (^|[^A-Za-z0-9_])git[[:space:]]+commit([^A-Za-z0-9_]|$) ]]; then
+  IS_COMMIT=1
+fi
+if [[ "$COMMAND" =~ (^|[^A-Za-z0-9_])git[[:space:]]+push([^A-Za-z0-9_]|$) ]]; then
+  IS_PUSH=1
+fi
+if [[ "$COMMAND" =~ (^|[^A-Za-z0-9_])gh[[:space:]]+pr[[:space:]]+create([^A-Za-z0-9_]|$) ]]; then
+  IS_PR=1
+fi
 
-if [[ "$COMMAND" =~ $BLOCK_REGEX ]]; then
+# Not a delegated operation — no policy applies
+if [ "$IS_COMMIT" = 0 ] && [ "$IS_PUSH" = 0 ] && [ "$IS_PR" = 0 ]; then
+  exit 0
+fi
+
+# ── Delegation check: only github-ops may run delegated operations ──
+AGENT_TYPE=$(echo "$INPUT" | jq -r '.agent_type // empty' 2>/dev/null || echo "")
+if [ "$AGENT_TYPE" != "f5xc-github-ops:github-ops" ]; then
   cat >&2 <<EOF
 BLOCKED: "${COMMAND}" is a delegated git/GitHub operation.
 
 CLAUDE.md requires all git commit, git push, and gh pr create calls to
 go through the f5xc-github-ops:github-ops subagent. Dispatch that agent
 with a clear task description instead of running the command directly.
+EOF
+  exit 2
+fi
+
+# ── Content check: governed files cannot leave a downstream repo ────
+SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
+GOVERNANCE_FILE="${SCRIPT_DIR}/../governance.json"
+if [ ! -f "$GOVERNANCE_FILE" ]; then
+  exit 0
+fi
+
+# Compute the set of file paths this operation would affect
+CHANGED=""
+if [ "$IS_COMMIT" = 1 ]; then
+  # Files staged for this commit
+  CHANGED=$(git diff --cached --name-only 2>/dev/null || echo "")
+  # -a / --all flag includes modified-but-unstaged tracked files
+  if [[ " $COMMAND " =~ [[:space:]]-[a-zA-Z]*a[a-zA-Z]*[[:space:]] ]] ||
+    [[ " $COMMAND " =~ [[:space:]]--all[[:space:]] ]]; then
+    CHANGED=$(printf '%s\n%s\n' "$CHANGED" "$(git diff --name-only 2>/dev/null || echo "")")
+  fi
+else
+  # push or pr create: files changed between upstream and HEAD
+  UPSTREAM=$(git rev-parse --abbrev-ref --symbolic-full-name '@{u}' 2>/dev/null || echo "")
+  if [ -z "$UPSTREAM" ]; then
+    UPSTREAM=$(git symbolic-ref --short refs/remotes/origin/HEAD 2>/dev/null || echo "origin/main")
+  fi
+  CHANGED=$(git diff --name-only "${UPSTREAM}..HEAD" 2>/dev/null || echo "")
+fi
+
+CHANGED_SORTED=$(printf '%s\n' "$CHANGED" | awk 'NF' | sort -u)
+GOVERNED_SORTED=$(jq -r '.protected_files[]' "$GOVERNANCE_FILE" 2>/dev/null | awk 'NF' | sort -u)
+
+if [ -z "$CHANGED_SORTED" ] || [ -z "$GOVERNED_SORTED" ]; then
+  exit 0
+fi
+
+VIOLATIONS=$(comm -12 <(printf '%s' "$CHANGED_SORTED") <(printf '%s' "$GOVERNED_SORTED"))
+
+if [ -n "$VIOLATIONS" ]; then
+  SOURCE_REPO=$(jq -r '.source_repo' "$GOVERNANCE_FILE")
+  cat >&2 <<EOF
+BLOCKED: "${COMMAND}" would affect governed file(s) managed by ${SOURCE_REPO}:
+$(printf '%s\n' "$VIOLATIONS" | sed 's/^/  - /')
+
+Governed files cannot be committed, pushed, or PR-submitted from
+downstream repos. To change them, open an issue/PR in ${SOURCE_REPO} —
+changes sync automatically to downstream repos.
 EOF
   exit 2
 fi

